django_loci/static/django-loci/js/loci.js

```diff
@@ -377,4 +497,120 @@ django.jQuery(function ($) {
+                    });
+            }
+            if (getMarker() !== undefined) {
+                if (getMarker().dragging._enabled === true) {
+                    window.update = true;
+                    updateAdress();
+                } else {
+                    window.beforeEditLat = currentMarkerLat;
+                    window.beforeEditLng = currentMarkerLng;
+                }
+            }
+        }, 500);
+
+        var submit = false;
+        $form.submit(function (e) {
+            var currentMarkerCoords = getMarker() !== undefined ? getMarker().getLatLng() : null;
+            var currentMarkerLat = getMarker() !== undefined ? getMarker().getLatLng().lat : null;
+            var currentMarkerLng = getMarker() !== undefined ? getMarker().getLatLng().lng : null;
+            if (currentMarkerCoords !== window.markerCoords && currentMarkerCoords !== null &&
+                    getMarker().dragging._enabled !== true) {
+                var reverseApiWithData = window.reverseApi + '?lat=' + currentMarkerLat + '&lng=' + currentMarkerLng;
+                window.fetch(reverseApiWithData)
+                    .then(function (resp) {
+                        var dataPromise = resp.json();
+                        dataPromise.then(function (data) {
+                            if ($addressInput.val()) {
+                                if (currentMarkerLat !== window.addressLat ||
+                                        currentMarkerLng !== window.addressLng) {
+                                    if (resp.status === 200) {
+                                        if (window.confirm('Would you like to update address?')) {
+                                            $addressInput.val(data.address);
+                                            submit = true;
+                                            $form.submit();
+                                        } else {
+                                            submit = true;
+                                            $form.submit();
+                                        }
+                                    } else {
+                                        var message = 'We couldn\'t find a related address to the location '
+                                                       + 'selected on the map, please enter the address manually';
+                                        alert(message);
+                                        window.skipGeocode = 2;
+                                    }
+                                } else {
+                                    submit = true;
+                                    $form.submit();
+                                }
+                            } else {
+                                $addressInput.val(data.address);
+                                submit = true;
+                                $form.submit();
+                            }
+                        });
+                    });
```

Comment:
```suggestion
        $form.submit(async function (e) {
            var currentMarkerCoords = getMarker() !== undefined ? getMarker().getLatLng() : null;
            var currentMarkerLat = getMarker() !== undefined ? getMarker().getLatLng().lat : null;
            var currentMarkerLng = getMarker() !== undefined ? getMarker().getLatLng().lng : null;
            if (currentMarkerCoords !== window.markerCoords && currentMarkerCoords !== null &&
                    getMarker().dragging._enabled !== true) {
                var reverseApiWithData = window.reverseApi + '?lat=' + currentMarkerLat + '&lng=' + currentMarkerLng;
                var resp = window.fetch(reverseApiWithData)
                if (!window.confirm('Would you like to update address?')) {
                    submit = true;
                    $form.submit();
                    return;
                }
                resp = await resp;
                var data = await resp.json();
                if ($addressInput.val()) {
                	if (currentMarkerLat !== window.addressLat ||
                	currentMarkerLng !== window.addressLng) {
                        if (resp.status === 200) {
                            $addressInput.val(data.address);
                            submit = true;
                            $form.submit();
                        } else {
                            var message = 'We couldn\'t find a related address to the location '
                                           + 'selected on the map, please enter the address manually';
                            alert(message);
                            window.skipGeocode = 2;
                        }
                    } else {
                        submit = true;
                        $form.submit();
                    }
                } else {
                    $addressInput.val(data.address);
                    submit = true;
                    $form.submit();
                }
```
Asynchronously do request to API, while waiting for user's confirmation.

Reply:
It doesn't work, but anyway thanks for making base for this feature :smile: